org.jacoco.core/src/org/jacoco/core/instr/Instrumenter.java

```diff
@@ -156,7 +146,11 @@ protected String getCommonSuperClass(final String type1,
 	 */
 	public void instrument(final InputStream input, final OutputStream output,
 			final String name) throws IOException {
-		output.write(instrument(input, name));
+		try {
+			output.write(instrument(input, name));
+		} catch (final RuntimeException e) {
+			throw instrumentError(name, e);
```

Comment:
IMO addition of this is an indication of regression in `instrument(InputStream, String)`.

Reply:
And if we don't wrap it with a try-catch block we got `Tests in error: testInstrumentAll_BrokenClassFileInZip(org.jacoco.core.instr.InstrumenterTest): 67` that is the ArrayOutOfBoundsException

Reply:
@andrioli if you fix the regression in `instrument(InputStream, String)` described in https://github.com/jacoco/jacoco/pull/600#discussion_r141427042 , then test won't be failing and try-catch won't be needed here.